### PR TITLE
docs: Adding notebook for ICE explainer

### PR DESCRIPTION
**Pull request by ezherdeva**

### CODE REVIEW

**Comment by ezherdeva** [comment]
@memoryz 
You didn't appear as a reviewer for some reason.

File: notebooks/Interpretability - ICE explainer.ipynb [hunk 1]
```diff
@@ -0,0 +1 @@
... easy to see the average."]},{"cell_type":"markdown","metadata":{"application/vnd.databricks.v1+cell":{"inputWidgets":{},"nuid":"b974545e-371b-41b0-b9a4-adb3036d9eb8","showTitle":false,"title":""}},"source":["## Summary\n","\n","Partial dependence plots (PDP) and Individual Conditional Expectation (ICE) plots can be used to visualize and analyze interaction between the target response and a set of input features of interest.\n","\n","Both PDPs and ICEs assume that the input features of interest are independent from the complement features, and this assumption is often violated in practice.\n","\n","ICE shows the dependence on average, but if you want to observe features individually - you can use ICE.\n","\n","Using examples above we showed how it can be usefull to draw such plots to analyze how machine learning model made their predictions, what was important and how we can interpret the results."]}],"metadata":{"application/vnd.databricks.v1+notebook":{"dashboards":[],"language":"python","notebookMetadata":{"pythonIndentUnit":2},"notebookName":"PDP-ICE-tutorial-new","notebookOrigID":2416290700869370,"widgets":{}},"language_info":{"name":"python"}},"nbformat":4,"nbformat_minor":0}
```
Comment (mhamilton723) [inline notebooks/Interpretability - ICE explainer.ipynb:1]: nit: Perhaps pump this through a json formatter so the diff lines doing forward will be nice

**Comment by mhamilton723** [comment]
pdp = ICETransformer(model=model, targetCol="probability", kind="average", targetClasses=[1]).\
    setCategoricalFeatures(categorical_features).\
    setNumericFeatures(numeric_features).setNumSamples(50)
    
   ^ Might want to use either setters or init args for consistency

**Comment by mhamilton723** [comment]
plt.title("")

Might not need this line if you don't want titles

**Comment by mhamilton723** [comment]
col_name =  'age_dependence'
age_dep = get_pandas_df_from_column(output, col_name)

Could remove some duplication with 
age_dep = get_pandas_df_from_column(output, 'age_dependence')

File: notebooks/Interpretability - ICE explainer.ipynb (hunk 1 above)
Comment (ezherdeva) [reply notebooks/Interpretability - ICE explainer.ipynb]: Can't find how to do it :( But you can hit the "View file" and it beautifully displays the file. The rest comments I fixed

**Comment by memoryz** [comment]
also please move the notebook to notebooks/features/responsible_ai folder.

**Comment by ezherdeva** [comment]
I've changed everything according to the comments. Also, I reformat the JSON view, it's easy to comment. Please, have a look

**Review by mhamilton723 — APPROVED** [review]

**Pull request merged**